src/sql/workbench/services/accountManagement/browser/accountDialog.ts

```diff
@@ -146,6 +147,7 @@ export class AccountDialog extends Modal {
 	public get onCloseEvent(): Event<void> { return this._onCloseEmitter.event; }
 
 	constructor(
+		private accountsLoaded: boolean,
```

Comment:
It makes more sense to me to have the accountDialog do the check itself for whether there's available providers when it gets rendered. So just injecting in the accountManagementService instead and checking the getAccountProviderMetadata list. 

Right now there's a small chance for a race condition if this is constructed but before it can register the listeners the account provider is registered. This means that accountsLoaded is false so it'll be waiting for accounts to be loaded, but that event was already fired so it'll hang forever.